sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -748,3 +748,18 @@ def _internal_resourcetype(resource_type: str) -> str:
     if resource_type.lower() ==  "partitionkey":
         return "colls"
     return resource_type
+
+def format_date_RFC_1123(dt: datetime) -> str:
```

Comment:
Can we add a few tests for this formatting function?

Reply:
In order to properly add tests for this in our test files, it would involve manually changing the locale which comes with three main issues: 1. Changing locale is OS dependent. 2. There is no guarantee any locale will be available without first being installed especially on lightweight Linux os distros. 3. Changing the locale on the fly might cause unforeseeable issues with all other tests since it changes it system wide. For now manually testing this seems to be the best approach. I have been able to test this with lux, french, thai, chinese, and spanish locales.

Reply:
Can we mock it? Pass different locale inputs and assert the expected format is returned?

Reply:
I just changed it to use a stdlib function that specifically deals with this issue.